cypress/integration/Input/Can_set_a_value.feature

```diff
@@ -0,0 +1,6 @@
+Feature: The Input can set a value
+
+    Scenario: The user types some text
+        Given a Input with no text is rendered
+        When the user types something
```

Comment:
I'd expect the input to also have to be focused. I see that we're `.get`ting the input in this step. To be more explicit about the expected behaviour, I'd include the following And:

`And the input is focused`

And remove the focusing from this step.

Reply:
I'm not sure on that one. That step is not really analogue to what is happening in the code:
```
When('the user types something', () => {
    cy.wrap('user input').as('userInput')
    cy.get('@userInput').then(userInput => {
        cy.get('input').type(userInput)
    })
})
```
So I don't think adding that "And the input is focussed" step is a good idea, since there is no need to set focus on the input before typing....

Reply:
So my reasoning here is that it seems to me that the test steps don't describe 100% how this would work. If the user types something, but the input isn't focused, it won't work.

So to make it more explicit, this is along the lines of what I'd expect:

<img width="586" alt="Screenshot 2019-12-10 at 14 56 25" src="https://user-images.githubusercontent.com/7355199/70535326-2ed76d00-1b5d-11ea-9df6-e5a40ed01857.png">

That way the test describes exactly how this feature is expected to work

Reply:
Discussed on slack, simplified this in https://github.com/dhis2/ui-forms/pull/108/commits/d39533b4a790e6544e9366117335c3f823665938

Reply:
I don't really agree with that. If we assume cypress is working correctly, then we should assume that the following command:
```javascript
 cy.get('input').type(userInput)
```
Does the following:
- Get the input
- Focus the input
- Simulate a keypress
- etc

When we add the step you suggest, I feel we are sort of manually recreating what `.type()` is already doing.

I've based my opinion mainly on the fact that `type()` is an "action command" according to the cypress docs which does lots of things to simulate "real" user behaviour: https://docs.cypress.io/api/commands/type.html#Actionability